web/modules/custom/dbcdk_community_content/dbcdk_community_content.tokens.inc

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * @file
+ * Builds placeholder replacement tokens for DBCDK Communtiy Content.
+ */
+
+use Drupal\Core\Render\BubbleableMetadata;
+
+/**
+ * Implements hook_token_info_alter().
+ *
+ * Add a sub-path token to nodes.
+ */
+function dbcdk_community_content_token_info_alter(array &$data) {
+  $data['tokens']['node']['sub-path'] = array(
+    'name' => t('Sub Path'),
+    'description' => t('The sub path of the article URL.'),
+  );
+}
+
+/**
+ * Implements hook_tokens().
+ *
+ * Give the node:sub-path token the aliased path for the taxonomy term it is
+ * referencing. - We need this because there is, current, no support for
+ * referenced entities "sub values".
+ */
+function dbcdk_community_content_tokens($type, array $tokens, array $data, array $options, BubbleableMetadata $bubbleable_metadata) {
+  $replacements = array();
+  if ($type === 'entity' & !empty($data['entity'])) {
+    /** @var \Drupal\Core\Entity\EntityInterface $entity */
+    $entity = $data['entity'];
+    foreach ($tokens as $name => $original) {
+      switch ($name) {
+        case 'sub-path':
```

Comment:
How about replacing `foreach/switch/case` with an `if ($tokens['sub-path'])`?

That seems a bit simpler than looping over all tokens.